test/integration/consul-container/test/README.md

```diff
@@ -0,0 +1,67 @@
+- Ensure that customers are able to upgrade to a new Consul version and new and existing features work as expected. 
+- Test for backwards compatibility. i.e support for upgrading directly from Consul `1.13` to `1.15`, without upgrading to `1.14`
+    - Highlights errors that will occur if attempting to skip a version during upgrades. Example the API changed between version 1.13 and 1.15 so direct upgrades between these version is not supported. 
+
+
+The tests focus on Consul upgrade lifecycle and emits logs to stdout which developers can be silenced. Please reach out to the [CTIA Team](https://hashicorp.slack.com/app_redirect?channel=team-consul-test-infra) for any questions or help getting started.
+
+## Getting Started
+We use [`golangci-lint`](https://golangci-lint.run/usage/install/), [`Makefile`](https://www.gnu.org/software/make/manual/make.html) and [`Docker`](https://docs.docker.com/get-docker/) in this repository. It's generally simpler to install the binaries rather than build from source. 
+
+### Running Upgrade integration tests
+To run integration tests locally,here are the commands needed:
+```sh
+make dev-docker
+cd ./test/integration/consul-container
+go test -v -timeout 30m -run ^TestACL_Upgrade_Node_Token$ ./.../upgrade/ --target-image consul --target-version local --latest-image consul --latest-version 1.15 -follow-log=false
+```
+
+The command above runs a single test. To run the entire upgrade test suite, the following command can be used.
+```sh
+go test -v -timeout 30m ./.../upgrade --target-image consul --target-version local --latest-image consul --latest-version 1.15 -follow-log=false
+```
+
+To follow the docker logs, remove `-follow-log=false` from the command above. 
+
+## Adding a new upgrade integration test  
+Upgrade integration tests are defined in the [test/integration/consul-container/test/upgrade](/test/integration/consul-container/test/upgrade) subdirectory and new upgrade integration tests should always be added to this location. The test framework uses
+[functional table-driven tests in Go](https://yourbasic.org/golang/table-driven-unit-test/) and using function types to modify the base value for each test case.
+
+For integration tests with multiple test cases, it should always start by invoking 
+```go
+	type testcase struct {
+		name string
+		create func()
+		extraAssertion func()
+	}
+```
+see example [here](./hashicorp/consul/test/integration/consul-container/test/upgrade/l7_traffic_management/resolver_default_subset_test.go). For upgrade tests with a single test case, they can be written like
+```go
+	run := func(t *testing.T, oldVersion, targetVersion string) {
+        // insert test
+    }
+	t.Run(fmt.Sprintf("Upgrade from %s to %s", utils.LatestVersion, utils.TargetVersion),
+		func(t *testing.T) {
+			run(t, utils.LatestVersion, utils.TargetVersion)
+		})
+```
+see example [here](./hashicorp/consul/test/integration/consul-container/test/upgrade/acl_node_test.go)
+
+### How it works
```

Comment:
I think visualization like the following will help developers establish a mental picture of the test scenario. Could you add a paragraph on how the test scenario is built.


<img width="549" alt="Screen Shot 2023-03-13 at 10 40 11 AM" src="https://user-images.githubusercontent.com/463631/224734653-a2f7f12b-afb9-4507-856b-32d6c9c6dc51.png">
rios.